text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,505 @@
+
+This core constuct directly creates a CloudFormation stack set vs. provisioning a single stack that deploys a stack set. 
+The deployment also updates the stack set, updates existing stack set instances, and monitors the stack set operation.
+
+### What is supported?
+
+* Self Manged Stack Sets
+  * Create, Update, and Update Existing Stack Set Instances directly through CDK
+
+### What is not supported?
+
+* Service Managed Stack Sets
+* Stack Set Instance Add/Remove through CDK
+* Stack Set deletion
+* CDK Assets (CDK File Assets/CDK Docker Assets)
+  * More information on CDK asset management and complexities described in the [File Assets Management](#4-file-assets-management) section.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+Today, CDK does not support the creation and updating of CloudFormation stack sets directly. This solution would also enable deploying to existing stack sets with low effort.
+
+CDK does provide the `cdk.CFNStackSet` construct, which could accomplish similar behavior. 
+The difference is that the `cdk.CFNStackSet` construct provisions a single CloudFormation stack with the resource type `AWS::CloudFormation::StackSet`, and then the stack provisions the actual stack set. 
+This requires a 1:1 mapping of parameters and tags that need to be passed from the stack to the stack set. 
+Customer can deploy to existing stack sets, but requires some migration steps for importing an existing stack set into the resource `AWS::CloudFormation::StackSet` before before being able to use `cdk.CFNStackSet`.
+
+### Why should we _not_ do this?
+
+We might not want to do this since CDK does provide the `cdk.CFNStackSet` construct, which could accomplish similar behavior. 
+The intermdiate stack may not be ideal for some customers, and may require large migration effort for customers who manage many stacksets.
+
+### What changes are required to enable this change?
+
+To enable this feature, we would need to introduce a new StackSet construct for customers to extend, a new assembly/artifact type for the Cloud Assembly, stack set synthesizer, and deployment code that creates/updates stack sets using AWS CloudFormation APIs.
+
+### Is this a breaking change?
+
+No.
+
+### What are the drawbacks of this solution?
+
+The handling of file assets is more complex than a normal stack CDK app since stack set instances are deployed independently from CDK context and exist in different accounts and regions. 
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CFNStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create Bootstrapless StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+
```

Comment:
I feel like we're missing the step where we add the calls to the CloudFormation APIs in the CLI to actually create/update the StackSets.